src/main/java/liqp/Template.java

```diff
@@ -589,6 +596,53 @@ public String renderUnguarded(TemplateContext parent) {
         return renderUnguarded(new HashMap<String, Object>(), parent, true);
     }
 
+    /**
+     * Returns the processed placeholder input string post parsing and rending.
+     *
+     * @param inputString: input string that contains 0 or multiple pre-processed placeholder strings.
+     * @param variables: contains the mapping of the placeholders and is used for
+     *                 mapping of the placeholder strings while rendering the input string.
+     */
+        public static String processPlaceHolderString(String inputString, Map<String, Object> variables) {
+            StringBuilder result = new StringBuilder();
+
+            int lastMatchEnd = 0;
+            Matcher matcher = null;
+            Pattern pattern;
+
+            try {
+                pattern = Pattern.compile("\\{\\{[^{}]*\\}\\}");
+                matcher = pattern.matcher(inputString);
+            } catch (Exception exception) {
+                System.err.println("Returning empty string. Exception occurred while searching for parsing placeholder strings: " + exception.getMessage());
```

Comment:
why System.err.println used everyhere and not existing logging framework

Reply:
this is liqp repo. it's not using slf4j.
we will take it slowly as it will require complete repo changes.